src/plugins/console/public/application/containers/editor/monaco/components/context_menu/context_menu.tsx

```diff
@@ -0,0 +1,260 @@
+} from '@elastic/eui';
+import { NotificationsSetup } from '@kbn/core/public';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { LanguageSelectorModal } from './language_selector_modal';
+import { convertRequestToLanguage } from '../../../../../../services';
+
+import { useServicesContext } from '../../../../../contexts';
+import { StorageKeys } from '../../../../../../services';
+import { DEFAULT_LANGUAGE } from '../../../../../../../common/constants';
+
+interface Props {
+  getRequests: () => Promise<any>;
+  getDocumentation: () => Promise<string | null>;
+  autoIndent: (ev: React.MouseEvent) => void;
+  notifications: NotificationsSetup;
+}
+
+const DELAY_FOR_HIDING_SPINNER = 500;
+
+export const ContextMenu = ({
+  getRequests,
+  getDocumentation,
+  autoIndent,
+  notifications,
+}: Props) => {
+  // Get default language from local storage
+  const {
+    services: { storage, esHostService },
+  } = useServicesContext();
+  const defaultLanguage = storage.get(StorageKeys.DEFAULT_LANGUAGE, DEFAULT_LANGUAGE);
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [isRequestConverterLoading, setRequestConverterLoading] = useState(false);
+  const [isLanguageSelectorVisible, setLanguageSelectorVisibility] = useState(false);
+  const [currentLanguage, setCurrentLanguage] = useState(defaultLanguage);
+
+  const copyText = async (text: string) => {
+    if (window.navigator?.clipboard) {
+      await window.navigator.clipboard.writeText(text);
+      return;
+    }
+    throw new Error('Could not copy to clipboard!');
+  };
+
+  // This function will convert all the selected requests to the language by
+  // calling convertRequestToLanguage for each request and then copy the data
+  // to clipboard.
+  const copyAs = async (language?: string) => {
+    // Get the language we want to convert the requests to
+    const withLanguage = (language || currentLanguage).toLowerCase();
+    // Get all the selected requests
+    const requests = await getRequests();
+
+    const { data: requestsAsCode, error: requestError } = await convertRequestToLanguage({
+      language: withLanguage,
+      esHost: esHostService.getHost(),
+      requests,
+    });
+
+    if (requestError) {
+      const error = new Error(requestError.message);
+      error.name = requestError.error;
+
+      notifications.toasts.addError(error, {
+        title: i18n.translate('console.consoleMenu.copyAsFailedMessage', {
+          defaultMessage:
+            '{requestsCount, plural, one {Request} other {Requests}} could not be copied to clipboard',
+          values: { requestsCount: requests.length },
+        }),
+      });
+
+      return Promise.reject(error);
+    }
+
+    notifications.toasts.addSuccess({
+      title: i18n.translate('console.consoleMenu.copyAsSuccessMessage', {
+        defaultMessage:
+          '{requestsCount, plural, one {Request} other {Requests}} copied to clipboard as {language}',
```

Comment:
According to the video and tooltips, the language name is lowercased here, is this expected?

Reply:
You got that right! Should be fixed now https://github.com/elastic/kibana/pull/189976/commits/e139abb3368daff460eac710112a448b7c3600aa